src/libs/ApiUtils.js

```diff
@@ -0,0 +1,86 @@
+import lodashGet from 'lodash/get';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../ONYXKEYS';
+import CONFIG from '../CONFIG';
+import CONST from '../CONST';
+import * as Environment from './Environment/Environment';
+
+// To avoid rebuilding native apps, native apps use production config for both staging and prod
+// We use the async environment check because it works on all platforms
+let ENV_NAME = CONST.ENVIRONMENT.PRODUCTION;
+let stagingServerToggleState = false;
+Environment.getEnvironment()
+    .then((envName) => {
+        ENV_NAME = envName;
+
+        // We connect here, so we have the updated ENV_NAME when Onyx callback runs
+        Onyx.connect({
+            key: ONYXKEYS.USER,
+            callback: (val) => {
+                const defaultToggleState = ENV_NAME === CONST.ENVIRONMENT.STAGING;
+                stagingServerToggleState = lodashGet(val, 'shouldUseStagingServer', defaultToggleState);
+            },
+        });
+    });
```

Comment:
Is it OK to run `Onyx.connect` in a callback instead of a top level call?
I've browsed existing code and found some similar usages

Reply:
I think if we are using it successfully in other places it should be fine.